src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.tsx

```diff
@@ -163,36 +135,31 @@ function BaseValidateCodeForm(props) {
 
     /**
      * Handle text input and clear formError upon text change
-     *
-     * @param {String} text
-     * @param {String} key
      */
-    const onTextInput = (text, key) => {
-        let setInput;
+    const onTextInput = (text: string, key: ValidateCodeFormVariant) => {
         if (key === 'validateCode') {
-            setInput = setValidateCode;
+            setValidateCode(text);
         }
         if (key === 'twoFactorAuthCode') {
-            setInput = setTwoFactorAuthCode;
+            setTwoFactorAuthCode(text);
         }
         if (key === 'recoveryCode') {
-            setInput = setRecoveryCode;
+            setRecoveryCode(text);
         }
 
-        setInput(text);
         setFormError((prevError) => ({...prevError, [key]: ''}));
```

Comment:
```suggestion
        setFormError((prevError) => ({...prevError, [key]: undefined}));
```

For consistency with https://github.com/Expensify/App/pull/35404#discussion_r1473040845